packages/front-end/components/Experiment/EditMetricsForm.tsx

```diff
@@ -38,9 +38,9 @@ const EditMetricsForm: FC<{
   const { hasCommercialFeature } = useUser();
   const hasOverrideMetricsFeature = hasCommercialFeature("override-metrics");
 
-  const { metrics: metricDefinitions, getDatasourceById } = useDefinitions();
+  const { metricsFilteredByProject, getDatasourceById } = useDefinitions();
```

Comment:
Yeah, the current project in the left nav should only really affect 2 things:
- Filter the list pages
- Set the initial project when creating new objects

Once an object is created, we need to use that object's project for all editing tasks.  It may not be the same thing as the project in the left nav.

Doing an intersection like you have should work.